core/embed/projects/prodtest/cmd/prodtest_secrets.c

```diff
@@ -141,15 +143,20 @@ static void prodtest_secrets_get_mcu_device_key(cli_t* cli) {
     return;
   }
 
-  ed25519_secret_key mcu_private = {0};
-  if (secret_key_mcu_device_auth(mcu_private) != sectrue) {
+  uint8_t seed[MLDSA_SEEDBYTES] = {0};
+  if (secret_key_mcu_device_auth(seed) != sectrue) {
     cli_error(cli, CLI_ERROR, "`secret_key_mcu_device_auth()` failed.");
-    return;
+    goto cleanup;
+  }
+
+  uint8_t mcu_public[CRYPTO_PUBLICKEYBYTES] = {0};
+  uint8_t mcu_private[CRYPTO_SECRETKEYBYTES] = {0};
+  if (crypto_sign_keypair_internal(mcu_public, mcu_private, seed) != 0) {
```

Comment:
Is this function specified in the standard, so we can be certain it won't change in the future?

Reply:
The NIST ML-DSA spec specifies key generation from a 32 byte seed, which is why I am even more surprised that the reference implementation does not expose it in its API.